hazelcast/src/main/java/com/hazelcast/instance/GroupProperty.java

```diff
@@ -0,0 +1,655 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance;
+
+import com.hazelcast.core.IMap;
+import com.hazelcast.internal.diagnostics.HealthMonitorLevel;
+import com.hazelcast.internal.metrics.ProbeLevel;
+import com.hazelcast.map.QueryResultSizeExceededException;
+import com.hazelcast.map.impl.query.QueryResultSizeLimiter;
+import com.hazelcast.query.TruePredicate;
+import com.hazelcast.query.impl.predicates.QueryOptimizerFactory;
+import com.hazelcast.spi.annotation.PrivateApi;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.util.Preconditions.checkHasText;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Defines the name and default value for Hazelcast properties.
+ *
+ * @deprecated since 3.7. Will be removed in 3.8.
+ */
+@PrivateApi
```

Comment:
The deprecated is already in the javadoc. 

I never understood the added value of the @Deprecated because with the javadoc version at least one can provide documentation about when it is deprecated.